datapipelines/src/test/java/io/customer/datapipelines/sdk/AndroidLifecyclePluginTests.kt

```diff
@@ -0,0 +1,296 @@
+import junit.framework.TestCase.assertFalse
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.runTest
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(manifest = Config.NONE)
+class AndroidLifecyclePluginTests {
+    private val lifecyclePlugin = AndroidLifecyclePlugin()
+
+    private lateinit var analytics: Analytics
+    private val mockContext = spyk(InstrumentationRegistry.getInstrumentation().targetContext)
+    private val mockApplication = mockk<Application>(relaxed = true)
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val testDispatcher = UnconfinedTestDispatcher()
+    private val testScope = TestScope(testDispatcher)
+
+    init {
+        setupMocks()
+    }
+
+    private fun setupMocks() {
+        val packageInfo = PackageInfo().apply {
+            versionCode = 100
+            versionName = "1.0.0"
+        }
+
+        val packageManager = mockk<PackageManager> {
+            every { getPackageInfo("com.foo", 0) } returns packageInfo
+        }
+        every { mockApplication.packageName } returns "com.foo"
+        every { mockApplication.packageManager } returns packageManager
+
+        mockkStatic(Instant::class)
+        every { Instant.now() } returns Date(0).toInstant()
+        mockkStatic(UUID::class)
+        every { UUID.randomUUID().toString() } returns String.random
+
+        mockHTTPClient()
+    }
+
+    private fun createTestAnalyticsInstance(moduleConfig: DataPipelinesModuleConfig): Analytics {
+        val configuration = createAnalyticsConfig(moduleConfig)
+        return object : Analytics(configuration, TestCoroutineConfiguration(testDispatcher, testScope)) {}
+    }
+
+    private fun createAnalyticsConfig(moduleConfig: DataPipelinesModuleConfig): Configuration {
+        return Configuration(writeKey = moduleConfig.cdpApiKey, mockApplication).let { config ->
+            updateAnalyticsConfig(moduleConfig = moduleConfig).invoke(config)
+            config
+        }
+    }
+
+    private fun createModuleInstance(
+        cdpApiKey: String = UnitTest.TEST_CDP_API_KEY,
+        applyConfig: CustomerIOBuilder.() -> Unit = {}
+    ): CustomerIO {
+        val builder = CustomerIOBuilder(mockContext as Application, cdpApiKey)
+        builder.setAutoAddCustomerIODestination(false)
+        builder.applyConfig()
+        return builder.build()
+    }
+
+    @Before
+    fun setup() {
+        analytics = createTestAnalyticsInstance(createModuleInstance().moduleConfig)
+    }
+
+    private fun setupAnalyticsForTest() {
+        analytics.configuration.trackApplicationLifecycleEvents = true
```

Comment:
Since we have its mapping in our config, I think we should be modifying this directly to make sure our config is working fine and we are not seeing false positives.

Reply:
removed it